### PR TITLE
2024 Q2 Clarity and bug fixes

### PR DESCRIPTION
**Pull request by lorcanrae**
Clarity and updates:

## SSH Keygen
- removed double line SSH keygen because it doesn't work on Windows and is generally annoying on unix-like.
- Added alternative if `ed25519` SSH key already exists (if alumni on unix-like or for professional reasons) 

fixes lewagon/data-engineering-setup#44

## Creating VM
- Updated `Machine Type` image reference for updated GCP console
- Added more clarity for boot image - included x86/64
- Updated `Reserve Static IP` image reference for updated GCP console
- Added more clarity for Windows and unix-like for copying pub SSH key

fixes lewagon/teachers#2199
fixes lewagon/data-engineering-setup#42

## VSCode
- Changed depreciated VSCode extension `better-toml` for `even-better-toml`
- Added note for remote-ssh config `Host` alias to not contain whitespace - known error

## Dotfiles
- Changed `gh repo clone lewagon/dotfiles` to `gh repo fork lewagon/dotfiles --clone` to ensure actual forking from LW's dotfiles (and the reason we now have branch protection on dotfiles).
- Added note about email selection and privacy mentioned in lewagon/teachers#2512 . Mirrored changes made in lewagon/setup#467

fixes lewagon/teachers#2213

## Docker
- Added hint box and solution if students encounter `permission denied while trying to connect to Docker daemon` when running `docker run hello-world`

## Python
- Added `pyenv-virtualenv` to install. `pyenv virtualenv-init` won't be removed from LW's `dotfiles` because it's required for Data Science and was giving an error on new terminal sessions. It is also useful for local python versioning.

## Spark
- Updated spark version from `3.5.0` to `3.5.1` so the curl doesn't 404

fixes lewagon/data-engineering-setup#45

### CODE REVIEW

**Comment by lorcanrae** [comment]
@gmanchon could you have a quick look at these if you have time. I'm usually happy to self-merge but I feel like someone at LW should be aware of the changes.

Did a fresh VM install from based on the updates and didn't encounter any issues :ok_hand:

File: _partials/docker.md [hunk 1]
```diff
@@ -20,4 +20,15 @@ docker info
 
 You should get:
 
+<details>
+  <summary markdown='span'>❗️ I recieved a permission denied when trying to connect to the Docker Daemon socket. </summary>
```
Comment (gmanchon) [inline _partials/docker.md:24]:
```suggestion
  <summary markdown='span'>❗️ I received a permission denied when trying to connect to the Docker Daemon socket. </summary>
```

File: _partials/docker.md [hunk 2]
```diff
@@ -20,4 +20,15 @@ docker info
 
 You should get:
 
+<details>
+  <summary markdown='span'>❗️ I received a permission denied when trying to connect to the Docker Daemon socket. </summary>
+
+If you recieved an error similar to the one below. navigate to the (GCP Compute Engine Console)[https://console.cloud.google.com/compute/instances] and STOP your VM (closing VSCode is not enough).
```
Comment (gmanchon) [inline _partials/docker.md:26]:
```suggestion
If you receive an error similar to the one below. navigate to the (GCP Compute Engine Console)[https://console.cloud.google.com/compute/instances] and STOP your VM (closing VSCode is not enough).
```

File: _partials/docker.md [hunk 3]
```diff
@@ -20,4 +20,15 @@ docker info
 
 You should get:
 
+<details>
+  <summary markdown='span'>❗️ I received a permission denied when trying to connect to the Docker Daemon socket. </summary>
+
+If you receive an error similar to the one below. navigate to the (GCP Compute Engine Console)[https://console.cloud.google.com/compute/instances] and STOP your VM (closing VSCode is not enough).
```
Comment (gmanchon) [inline _partials/docker.md:26]:
```suggestion
If you receive an error similar to the one below, navigate to the (GCP Compute Engine Console)[https://console.cloud.google.com/compute/instances] and STOP your VM (closing VSCode is not enough).
```

File: _partials/ubuntu_docker.md [hunk 4]
```diff
@@ -38,6 +38,16 @@ newgrp docker
 
 Run `docker run hello-world`, you should see something like:
 
+<details>
+  <summary markdown='span'>❗️ Permission denied while trying to connect to the Docker daemon socket. ❗️ </summary>
+
+If you received an error similar to the one below. navigate to the [GCP Compute Engine Console](https://console.cloud.google.com/compute/instances) and shut down your VM by selecting the tick box next to your VM instance and clicking STOP (closing and reopening VSCode is not enough).
```
Comment (gmanchon) [inline _partials/ubuntu_docker.md:44]:
```suggestion
If you receive an error similar to the one below, navigate to the [GCP Compute Engine Console](https://console.cloud.google.com/compute/instances) and shut down your VM by selecting the tick box next to your VM instance and clicking STOP (closing and reopening VSCode is not enough).
```

File: _partials/ssh_key.md [hunk 5]
```diff
@@ -13,31 +13,44 @@ We highly recommend installing [Windows Terminal](https://apps.microsoft.com/sto
...
 
 ℹ️ Don't worry if nothing prompt when you type, that is perfectly normal for security reasons.
 
 - You should be asked to `Enter same passphrase again`, do it.
 
 **❗️ You must remember this passphrase.**
 
+<details>
+  <summary markdown='span'> ❗️ /home/your_username/.ssh/id_ed25519 already exists.</summary>
+If you receive this message, you may already have an SSH Key with the same name (if you are a Le Wagon Alumni or you using SSH Authentication with Github).
```
Comment (gmanchon) [inline _partials/ssh_key.md:46]:
```suggestion
If you receive this message, you may already have an SSH Key with the same name (if you are a Le Wagon Alumni or are using SSH Authentication with Github).
```

**Review by gmanchon — APPROVED** [review]
LGTM 👌 Nice work @lorcanrae, thanks for the ping 🙏

**Pull request merged**